src/modules/paging/all.js

```diff
@@ -1,24 +1,39 @@
 import { flatten, range, isNil } from 'lodash';
-import { getConfig } from '@globality/nodule-config';
+import { getConfig, getContainer } from '@globality/nodule-config';
 import { MaxLimitReached } from '../../error';
 import concurrentPaginate from '../concurrency';
 
 const DEFAULT_LIMIT = 20;
+const DEFAULT_PAGING_UPPER_BOUND = 200;
 
 export default async function all(
     req,
     { searchRequest, args = {}, maxLimit = null, concurrencyLimit = 1 },
 ) {
     const { limit, offset, ...searchArgs } = args;
 
-    const defaultLimit = getConfig('defaultLmit') || DEFAULT_LIMIT;
+    const defaultLimit = getConfig('defaultLimit') || DEFAULT_LIMIT;
 
     const params = {
         ...searchArgs,
         limit: limit || defaultLimit,
         offset: offset || 0,
     };
     const firstPage = await searchRequest(req, params);
+
+    if (limit === defaultLimit && firstPage.count > DEFAULT_PAGING_UPPER_BOUND) {
+        const { logger } = getContainer();
+        logger.warning(
+            req,
+            'A large dataset requested with default limit set.',
```

Comment:
This error message confused me a little at first. As we didn't request a large dataset in this case right, we asked for a `limit=20` and got a large response in spite of this? 🤔 

A different wording might help like, 'A large dataset was returned ...', instead.

Reply:
It's a big dataset because you are using `all` function to automatically follow the paging and retrieve all the results. 

In this case the dataset refers to all the total results you are getting from the query (`count` in our pagination schema), where as `limit` is the number of results per page. So even though the page is small the number you requests `all` will do for you to get the full dataset is high 200 / 20 = 10 requests the service.

Reply:
Aha I see, yeah I was thinking it was the page result size. Thanks 👍